packages/pigeon/lib/dart_generator.dart

```diff
@@ -204,6 +204,10 @@ class DartGenerator extends StructuredGenerator<DartOptions> {
   ) {
     void writeValueDecode(NamedType field, int index) {
       final String resultAt = 'result[$index]';
+      String castCallPrefix = field.type.isNullable ? '?' : '!';
```

Comment:
Why pull this out when it's different in the two places it's used? Keeping it local in each spot, and final, seems less confusing.